lib/src/test/java/io/ably/lib/test/realtime/RealtimeResumeTest.java

```diff
@@ -524,4 +524,113 @@ public void resume_verify_publish() {
+			assertEquals(
+			    "The sender's channel should be attached!",
+			    senderChannel.state, ChannelState.attached
+			);
+
+			/* create and attach channel to recv on */
+			final Channel receiverChannel = receiver.channels.get(channelName);
+			receiverChannel.attach();
+			(new ChannelWaiter(receiverChannel)).waitFor(ChannelState.attached);
+			assertEquals(
+				"The receiver's channel should be attached!",
+				receiverChannel.state, ChannelState.attached
+			);
+			/* subscribe */
+			MessageWaiter messageWaiter =  new MessageWaiter(receiverChannel);
+
+			/* publish first messages to the channel */
+			CompletionSet msgComplete1 = new CompletionSet();
+			for(int i = 0; i < messageCount; i++) {
+				senderChannel.publish("test_event", "Test message (resume_publish_queue) " + i, msgComplete1.add());
+				try { Thread.sleep(delay); } catch(InterruptedException e){}
+			}
+
+			/* wait for the publish callback to be called */
+			ErrorInfo[] errors = msgComplete1.waitFor();
+			assertTrue(
+			    "First round of messages has errors!", errors.length == 0
+			);
+
+			/* wait for the subscription callback to be called */
+			messageWaiter.waitFor(messageCount);
+			assertEquals(
+			    "Did not receive the entire first round of messages!",
+			    messageWaiter.receivedMessages.size(), messageCount
+			);
+			messageWaiter.reset();
+
+			/* disconnect the sender, without closing;
+			 * NOTE this depends on knowledge of the internal structure
+			 * of the library, to simulate a dropped transport without
+			 * causing the connection itself to be disposed */
+			sender.connection.connectionManager.requestState(ConnectionState.disconnected);
+
+			/* wait */
+			try { Thread.sleep(2000L); } catch(InterruptedException e) {}
+
+			/*
+			 *  publish further messages to the channel, which should be queued
+			 *  because the channel is currently disconnected.
+			 */
+			CompletionSet msgComplete2 = new CompletionSet();
+			for(int i = 0; i < messageCount; i++) {
+				senderChannel.publish("test_event", "Test queued message (resume_publish_queue) " + i, msgComplete2.add());
+				try { Thread.sleep(delay); } catch(InterruptedException e){}
+			}
+
+			/* reconnect the sender */
+			sender.connection.connect();
+			(new ConnectionWaiter(sender.connection)).waitFor(ConnectionState.connected);
+
+
+			/* wait for the publish callback to be called.*/
+			errors = msgComplete2.waitFor();
+			assertTrue(
+			    "Second round of messages (queued) has errors!",
+				errors.length == 0
+			);
+
+			/* wait for the subscription callback to be called */
+			messageWaiter.waitFor(messageCount);
+			assertEquals(
+			    "Did not receive the entire second round of messages (queued)!",
+				messageWaiter.receivedMessages.size(), messageCount
+			);
+		} catch (AblyException e) {
+			e.printStackTrace();
+			fail("init0: Unexpected exception instantiating library");
+		} finally {
+			if(sender != null)
```

Comment:
As a point of style, I'm not really a fan of a single-line if clause without braces. There are some instances of this in the library and the tests for historical reasons, and we should try to fix them if a file is being modified as part of other work.